From c38b8f2fc22a2c220f65eb2f347f6f0a0219ab3c Mon Sep 17 00:00:00 2001
From: Weiwen Chen <cww@rock-chips.com>
Date: Mon, 10 Aug 2020 17:06:15 +0800
Subject: [PATCH] ARM: dts: rv1126: 38x38 board: fix flash led use pwm

Test:
insmod kernel/drivers/leds/leds-pwm.ko
echo 77 > /sys/devices/platform/pwmleds/leds/PWM-IR/brightness

Signed-off-by: Weiwen Chen <cww@rock-chips.com>
Change-Id: I4dd75bd124eeed1b2e13c638c34d315772aea3c1
---
 arch/arm/boot/dts/rv1126-38x38-v10-spi-nor.dts | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rv1126-38x38-v10-spi-nor.dts b/arch/arm/boot/dts/rv1126-38x38-v10-spi-nor.dts
index be9b5f4b83b3..666ec8d86caa 100644
--- a/arch/arm/boot/dts/rv1126-38x38-v10-spi-nor.dts
+++ b/arch/arm/boot/dts/rv1126-38x38-v10-spi-nor.dts
@@ -18,6 +18,16 @@
 	/delete-node/ vdd-npu;
 	/delete-node/ vdd-vepu;
 
+	pwmleds {
+		compatible = "pwm-leds";
+
+		pwmi_ir {
+			label = "PWM-IR";
+			pwms = <&pwm3 0 25000 0>;
+			max-brightness = <255>;
+		};
+	};
+
 	vcc_1v8: vcc-1v8 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_1v8";
@@ -101,7 +111,6 @@
 		compatible = "rockchip,ircut";
 		ircut-open-gpios = <&gpio3 RK_PB0 GPIO_ACTIVE_HIGH>;
 		ircut-close-gpios  = <&gpio3 RK_PB1 GPIO_ACTIVE_HIGH>;
-		led-gpios = <&gpio0 RK_PC1 GPIO_ACTIVE_HIGH>;
 		rockchip,camera-module-index = <1>;
 		rockchip,camera-module-facing = "front";
 	};
@@ -267,6 +276,12 @@
 	vccio7-supply = <&vcc3v3_sys>;
 };
 
+&pwm3 {
+	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm3m0_pins_pull_down>;
+};
+
 &rkisp {
 	status = "okay";
 
-- 
2.35.3

